cbs/settings.py

```diff
@@ -19,35 +19,53 @@ def __getattribute__(self, name):
         return val
 
     @classmethod
-    def use(cls, env="DJANGO_MODE"):
+    def use(cls, env="DJANGO_MODE", env_path=None):
         """Helper for accessing sub-classes via env var name.
 
         Gets a sub-class instance using ``get_settings_instance``, and returns
         the results of calling ``getattr__factory`` and ``dir_factory`` on it.
 
-        :param str env: Envirionment variable to get settings mode name from.
+        :param str env: Environment variable to get settings mode name from.
+        :param str env_path: the path to the file containing the environment variable, if left empty, os.environ will be used instead.
         :return: functions suitable for module-level ``__getattr__`` and
             ``__dir__``
         """
-        settings = cls.get_settings_instance(env)
+        settings = cls.get_settings_instance(env, env_path)
 
         return (
             settings.getattr_factory(),
             settings.dir_factory(),
         )
 
     @classmethod
-    def get_settings_instance(cls, env="DJANGO_MODE"):
+    def get_settings_instance(cls, env="DJANGO_MODE", env_path=None):
         """Create an instance of the appropriate Settings sub-class.
 
-        Takes the value of ``os.environ[env]``, calls ``.title()`` on it, then
-        appends `"Settings"`.
+        Takes the value of ``os.environ[env]``, calls ``.title()`` on it,
+        or reads the value from the specified file in ``env_path``,
+        then appends `"Settings"`.
 
         It will then find a sub-class of that name, and return an instance of
         it.
 
         """
         base = os.environ.get(env, "")
+
+        if env_path:
+            try:
+                import eniron
```

Comment:
Is this a typo? You reference `environ` below.

Reply:
ah
yes it should be environ
not sure how my ide didn't catch this 🫤